Source/Scene/PointCloudPostProcessor.js

```diff
@@ -0,0 +1,1178 @@
+/*global define*/
+define([
+        '../Core/Color',
+        '../Core/combine',
+        '../Core/ComponentDatatype',
+        '../Core/defined',
+        '../Core/destroyObject',
+        '../Core/Geometry',
+        '../Core/GeometryAttribute',
+        '../Core/PixelFormat',
+        '../Core/PrimitiveType',
+        '../Renderer/BufferUsage',
+        '../Renderer/ClearCommand',
+        '../Renderer/DrawCommand',
+        '../Renderer/Framebuffer',
+        '../Renderer/Pass',
+        '../Renderer/PixelDatatype',
+        '../Renderer/RenderState',
+        '../Renderer/Sampler',
+        '../Renderer/ShaderSource',
+        '../Renderer/ShaderProgram',
+        '../Renderer/Texture',
+        '../Renderer/TextureMagnificationFilter',
+        '../Renderer/TextureMinificationFilter',
+        '../Renderer/TextureWrap',
+        '../Renderer/VertexArray',
+        '../Scene/BlendEquation',
+        '../Scene/BlendFunction',
+        '../Scene/BlendingState',
+        '../Scene/StencilFunction',
+        '../Scene/StencilOperation',
+        '../Shaders/PostProcessFilters/PointOcclusionPassGL1',
+        '../Shaders/PostProcessFilters/PointOcclusionPassGL2',
+        '../Shaders/PostProcessFilters/RegionGrowingPassGL1',
+        '../Shaders/PostProcessFilters/RegionGrowingPassGL2',
+        '../Shaders/PostProcessFilters/DensityEdgeCullPass',
+        '../Shaders/PostProcessFilters/PointCloudPostProcessorBlendPass'
+    ], function(
+        Color,
+        combine,
+        ComponentDatatype,
+        defined,
+        destroyObject,
+        Geometry,
+        GeometryAttribute,
+        PixelFormat,
+        PrimitiveType,
+        BufferUsage,
+        ClearCommand,
+        DrawCommand,
+        Framebuffer,
+        Pass,
+        PixelDatatype,
+        RenderState,
+        Sampler,
+        ShaderSource,
+        ShaderProgram,
+        Texture,
+        TextureMagnificationFilter,
+        TextureMinificationFilter,
+        TextureWrap,
+        VertexArray,
+        BlendEquation,
+        BlendFunction,
+        BlendingState,
+        StencilFunction,
+        StencilOperation,
+        PointOcclusionPassGL1,
+        PointOcclusionPassGL2,
+        RegionGrowingPassGL1,
+        RegionGrowingPassGL2,
+        DensityEdgeCullPass,
+        PointCloudPostProcessorBlendPass
+    ) {
+    'use strict';
+
+     /**
+     * @private
+     */
+    function PointCloudPostProcessor(options) {
+        this._framebuffers = undefined;
+        this._colorTextures = undefined;
+        this._ecTextures = undefined;
+        this._densityTexture = undefined;
+        this._edgeCullingTexture = undefined;
+        this._sectorLUTTexture = undefined;
+        this._aoTextures = undefined;
+        this._stencilMaskTexture = undefined;
+        this._drawCommands = undefined;
+        this._clearCommands = undefined;
+
+        this.densityScaleFactor = '10.0';
```

Comment:
This can be a number instead of a string.

Reply:
It's slightly easier for it to remain a string because javascript keeps representing the number `10.0` as `10` when it is stringified. When the preprocessor define gets copied in, there are casting issues.